nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -237,16 +243,25 @@ WaypointFollower::followWaypoints()
       RCLCPP_INFO(
         get_logger(), "Task execution at waypoint %i %s", goal_index,
         is_task_executed ? "succeeded" : "failed!");
+
+      if (!is_task_executed) {
+        missed_waypoints_.emplace_back(MissedWaypoint{goal_index, current_goal_status_.error_code});
```

Comment:
How is this status code set to the missed waypoint code from the message enum? Shouldn't this be manually set to the WPF error code? Or is this changed somewhere I missed?